### PR TITLE
All RSVPed Guests See The List of RSVPs

### PR DESCRIPTION
**Pull request by Kanika1799**

### CODE REVIEW

File: src/components/EventPage/RsvpSection.tsx [hunk 1]
```diff
@@ -154,6 +176,33 @@ export const SessionsWrapper = ({
...
         </div>
+        {sortedSessions.map((session, key) => {
+          const userRsvp = session.rsvps.some(
+            (rsvp) => rsvp.attendeeId === user.id
+          );
+          return (
+            <>
+              {userRsvp && (
+                <details className="group">
+                  <summary className="flex cursor-pointer list-none items-center justify-between font-medium">
```
Comment (anggxyz) [inline src/components/EventPage/RsvpSection.tsx:187]: this should be outside the `map`

File: src/components/EventPage/RsvpSection.tsx [hunk 2]
```diff
@@ -154,6 +176,33 @@ export const SessionsWrapper = ({
...
+                    See All Rsvps list <TransitioningArrow />
```
Comment (anggxyz) [inline src/components/EventPage/RsvpSection.tsx:188]: this should be outside the `map`. is displaying the accordion as many times as the number of rsvps
image

**Comment by anggxyz** [comment]
can we use the component we already have? so it's easier to update/fix later
image

**Comment by anggxyz** [comment]
ty for the PR btw! ❤️ I appreciate you working on the issue :)

File: src/components/EventPage/RsvpSection.tsx [hunk 3]
```diff
@@ -155,6 +157,7 @@ export const SessionsWrapper = ({
             {Intl.DateTimeFormat().resolvedOptions().timeZone}
           </span>
         </div>
+        <SessionsDetailsNonSubmittable sessions={sessions} />
```
Comment (anggxyz) [inline src/components/EventPage/RsvpSection.tsx:160]: this needs to be inside a check that checks for currently logged in user. if the user is signed out the component should not be visible. then it needs to check if the user has already RSVPd to one of the events in the current convo. only then the component should be visible

Comment (anggxyz) [reply src/components/EventPage/RsvpSection.tsx:160]: and if the checks pass, this component should replace the above component with an option to remove RSVP if the user wants

**Review by anggxyz — CHANGES_REQUESTED** [review]

**Comment by avirajkhare00** [comment]
Great PR

**Pull request merged**